wp-config-sample.php

```diff
@@ -81,9 +81,30 @@
 
 /* That's all, stop editing! Happy blogging. */
 
+/**
+ * Allow developer to specify that ClassicPress core is stored
+ * somewhere else besides the root, such as '/classicpress',
+ * '/cp', '/core', or elsewhere. To use this feature define()
+ * `CORE_PATH` at the top of this file.
+ */
+if ( ! defined( 'CORE_PATH' ) ) {
+	define( 'CORE_PATH',  '/' );
+}
+
+/**
+ * Allow developer to specify that the ClassicPress "content"
+ * directory is located somewhere else besides '/wp-content',
+ * such as '/content', '/app', or other. To use this feature define()
+ * `CONTENT_PATH` at the top of this file.
+ */
+if ( ! defined( 'CONTENT_PATH' ) ) {
+	define( 'CONTENT_PATH',  '/wp-content' );
+}
+
 /** Absolute path to the ClassicPress directory. */
-if ( !defined('ABSPATH') )
-	define('ABSPATH', dirname(__FILE__) . '/');
+if ( ! defined( 'ABSPATH' ) ) {
+	define( 'ABSPATH', dirname( __FILE__ ) . CORE_PATH );
+}

```

Comment:
As written, the `CORE_PATH` constant here doesn't do much of anything.

`wp-load.php` sets `ABSPATH` to its containing directory:

https://github.com/ClassicPress/ClassicPress/blob/feac6fb1a9f5e9d2ee8a787185cf5c837755bf13/src/wp-load.php#L19-L22

Then it loads `wp-config.php` from either the same directory or the parent directory:

https://github.com/ClassicPress/ClassicPress/blob/feac6fb1a9f5e9d2ee8a787185cf5c837755bf13/src/wp-load.php#L26-L44

So the check in `wp-config.php` (copied from `wp-config-sample.php` for new installs) will never be executed under normal, correct loading behavior (via `wp-load.php`).

If we need to include `CORE_PATH` then we need to think through the desired logic a bit more (I think this would mean modifying `wp-load.php` to be able to find `wp-config.php` in more situations, but I am not clear on how).

Use of the `CORE_PATH` constant in `index.php` has also been mentioned - I'd like to know more about this.

Reply:
@nylen Originally I submitted this as part of a larger pull request where together the use-cases were clearer.  Broken out it is harder to see the value. 

The intention for breaking `CORE_PATH` out was:

1. To change `wp-config.php` so that all changes could be made to the top of the file in order to discourage changed at the bottom half of the file. 
 Ideally `wp-config-sample.php` would not have anything pre-coded except for them absolutely required such as `DB_NAME`, `DB_USER` and `DB_PASSWORD` plus the salts.  But with WordPress you get people needing to make changes throughout the file, so everything becomes inconsistent, especially when web hosts create their own mods you can't change.

2. To decouple the `ABSPATH` from the web root directory. I have found numerous times in custom coding where I needed to know the core path and it was always fragile hack to extract it out of `ABSPATH`.  Also, with `CORE_PATH` I can find the root direct without any fragileness, and WordPress does not make that direct or easy.

3. In the legacy case `ABSPATH` is used to _(find and to)_ load `wp-settings.php`, so it needs to be set in `wp-config.php`, unless we ignored legacy concerns.

4. Ideally `CORE_PATH` would be set and available in `index.php` because otherwise if you move where WordPress is located you have to edit that file, which is hacky.  That is actually what Better WP-Config make possible, but I did not find a way to do so with hard-coded constants without having to tell the site builder to edit `index.php`, which was what I wanted to avoid being a requirement.

> _"So the check in `wp-config.php` (copied from `wp-config-sample.php` for new installs) will never be executed under normal, correct loading behavior (via wp-load.php)."_

I don't follow this.  Which check in `wp-config.php`?